File: 백준/Gold/1300. K번째 수/K번째 수.cc
#include <stdio.h>
#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
	int n, k;
	int cnt;
	cin >> n >> k;
	int l = 1, r = k, mid;
	while (l <= r) {
		mid = (l + r) / 2;
		cnt = 0;
		for (int i = 1; i <= n; i++)
			cnt += min(mid / i, n);
		if (cnt >= k) r = mid - 1;
		else l = mid + 1;
	}
	cout << l;
	return 0;
}

File: 백준/Gold/1300. K번째 수/K번째 수.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int n = Integer.parseInt(br.readLine());
		int k = Integer.parseInt(br.readLine());
		int l = 1, r = k, mid, cnt;
		while(l <= r) {
			mid = (l + r) / 2;
			cnt = 0;
			for(int i = 1; i <= n; i++)
				cnt += Math.min(n, mid / i);
			if(cnt < k)
				l = mid + 1;
			else
				r = mid - 1;
		}
		System.out.println(r + 1);
	}
}